routes/api.php

```diff
@@ -27,6 +29,10 @@
     Route::resource('tag', TagController::class)->except(['create', 'edit']);
     Route::resource('sender-identity', SenderIdentityController::class)->except(['create', 'edit']);
     Route::resource('campaign', CampaignController::class)->except(['create', 'edit']);
+    Route::resource('role', RoleController::class)->except(['create', 'edit']);
+    Route::resource('user/user', UserController::class)->except(['create', 'edit']);
+    Route::post('user/update-user-role', [UserController::class, 'updateUserRoles'])->name('user.updateUserRoles');
+    Route::post('role/update-role-permission', [RoleController::class, 'updateRolePermissions'])->name('role.updateRolePermissions');
```

Comment:
Once you have route params, these routes should be named as:
```suggestion
    Route::post('user/{user}/role', [UserController::class, 'updateUserRoles'])->name('user.update-roles');
    Route::post('role/{role}/permission', [RoleController::class, 'updateRolePermissions'])->name('role.update-permissions');
```

Also you can simply the names as `user.updateRoles` as the word `user` is already present.

Let's keep the route names as hyphen-separated